src/lib/net/util/identifiers.ts

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable roblox-ts/lua-truthiness */
+
+import { RunService } from "@rbxts/services";
+
+export namespace Identifiers {
+	let autoSerde: Folder | undefined;
+	let numOfSerials = 0;
+
+	const limit = 65536;
+
+	const receiveDict: { [x: string]: unknown } = {};
+	const sendDict: { [x: string]: string | undefined } = {};
+
+	const isServer = RunService.IsServer();
+
+	/** @hidden */
+	export function init(): void {
+		if (!isServer) {
+			autoSerde = script.WaitForChild("auto_serde") as Folder;
+
+			for (const [id, value] of autoSerde.GetAttributes()) {
+				sendDict[id] = value as string;
+				receiveDict[value as never] = id;
+			}
+
+			autoSerde.AttributeChanged.Connect((id: string) => {
+				const packed = autoSerde?.GetAttribute(id) as string;
+
+				if (packed !== undefined) {
+					sendDict[id] = packed;
+					receiveDict[packed as never] = id;
+				} else {
+					const old = sendDict[id];
+
+					sendDict[id] = undefined;
+					receiveDict[old as never] = undefined;
+				}
+			});
+		} else {
+			autoSerde = new Instance("Folder");
+			autoSerde.Name = "auto_serde";
+			autoSerde.Parent = script;
+		}
+	}
+
+	/**
+	 * Creates a unique identifier for an endpoint, if it's on client it awaits for the result.
+	 *
+	 * @server
+	 *
+	 * @param id an id to compress from.
+	 */
+	export function create(): string {
+		const current = ++numOfSerials;
+		const id = tostring(current);
+
+		if (current >= limit) {
+			throw `Over the identification cap ${id}`;
+		}
+
+		if (!sendDict[id] && !isServer) {
```

Comment:
Comment this so it's less confusing please.

Reply:
For sure.